sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -288,7 +289,7 @@ object FileFormatWriter extends Logging {
         val escaped = ScalaUDF(
           ExternalCatalogUtils.escapePathName _,
           StringType,
-          Seq(Cast(c, StringType)),
+          Seq(Cast(c, StringType, DateTimeUtils.defaultTimeZone().getID)),
```

Comment:
This seems related to partition values.
We'll send follow-up prs about partition values ([SPARK-18939](https://issues.apache.org/jira/browse/SPARK-18939)), let's discuss there.